CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+cmake_minimum_required(VERSION 3.16)
+
+set(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")
+
+project(FreeRTOS-Kernel)
+
+list(APPEND CMAKE_MESSAGE_CONTEXT "FreeRTOS-Kernel")
+
+add_library(FreeRTOS-Kernel OBJECT)
+
+# Optional parameters.
+option(FREERTOS_KERNEL_ENABLE_COROUTINE "Enable co-routines by compiling croutine.c as part of the kernel source files." ON)
+option(FREERTOS_KERNEL_ENABLE_EVENT_GROUPS "Enable event groups by compiling event_groups.c as part of the kernel source files." ON)
+option(FREERTOS_KERNEL_ENABLE_STREAM_BUFFER "Enable stream and message buffers by compiling stream_buffer.c as part of the kernel source files." ON)
+option(FREERTOS_KERNEL_ENABLE_TIMER "Enable timers by compiling timers.c as part of the kernel source files." ON)
+
+option(FREERTOS_KERNEL_USE_CUSTOM_PORT "Enable the use of a custom port implementation from a different project. When enabling this option set <FREERTOS_KERNEL_PORT> to the path that contains port files." OFF)
+option(FREERTOS_KERNEL_USE_CUSTOM_HEAP "Enable the use of a custom heap implementation from a different project. When enabling this option set <FREERTOS_KERNEL_HEAP_FILE> to the file that implements the memory management." OFF)
```

Comment:
The intention here wasn't to set the `configUSE_TIMERS` directive, but to give the user the option to omit compiling unnecessary source files.

In practice most projects probably compile these sources whether they use them or not which is why they default to `ON`. If someone were to `set(FREERTOS_KERNEL_ENABLE_TIMER OFF)` and `#define configUSE_TIMERS    (1)` in their FreeRTOSConfig.h they would get a linker error when trying to use timer functions.

These options could be removed if they're more confusing than useful.

Reply:
I see, I personally did not like the idea that I'd need to make a change at 2 locations to add/remove any option to my build. @kstribrnAmzn - your thoughts?
I would like it better if it is at one location only - creating less confusion.

Reply:
> I see, I personally did not like the idea that I'd need to make a change at 2 locations to add/remove any option to my build.

I agree, does seem strange.

Reply:
> I would like it better if it is at one location only - creating less confusion.

Since `configUSE_TIMERS`, `configUSE_CO_ROUTINES`, etc define out the entirety of the source file there's no need for these options. They can be removed.